core/src/codegen/from_meta_impl.rs

```diff
@@ -90,16 +89,37 @@ impl<'a> ToTokens for FromMetaImpl<'a> {
                     }
                 };
 
+                let default_or_err = base.default.as_ref().map_or(
```

Comment:
Nit: I'd write this as a `map` call followed by an `unwrap_or_else` call, so that it's easier to see we transform `default_expr` if set, and then unwrap or create the error.